src/Monolog/Formatter/ElasticsearchLogstashHandler.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace AppBundle\Monitoring\Monolog\Handler;
+
+use Monolog\Formatter\FormatterInterface;
+use Monolog\Formatter\LogstashFormatter;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Logger;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Push logs directly to Elasticsearch and format them according to Logstash specification
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class ElasticsearchLogstashHandler extends AbstractProcessingHandler
+{
+    private $endpoint;
+    private $index;
+    private $client;
+    private $responses;
+
+    public function __construct(string $endpoint = 'http://elasticsearch:9200', string $index = 'monolog', HttpClientInterface $client = null, int $level = Logger::DEBUG, bool $bubble = true)
+    {
+        if (!class_exists(HttpClientInterface::class)) {
+            throw new \LogicException(sprintf('%s handler needs symfony/http-client, please run `composer require symfony/http-client`.', __CLASS__));
+        }
+
+        parent::__construct($level, $bubble);
+        $this->endpoint = $endpoint;
+        $this->index = $index;
+        $this->client = $client ?: HttpClient::create();
+        $this->responses = new \SplObjectStorage();
+    }
+
+    protected function write(array $record)
+    {
+        $this->sendToElasticsearch(array($record));
+    }
+
+    protected function getDefaultFormatter()
+    {
+        return new LogstashFormatter('application', null, null, 'ctxt_', LogstashFormatter::V1);
+    }
+
+    public function handleBatch(array $records)
+    {
+        $this->sendToElasticsearch($records);
+    }
+
+    private function sendToElasticsearch(array $documents)
+    {
+        $body = '';
+        foreach ($documents as $document) {
+            $body .= '{"index":{"_index":"'.$this->index.'","_type":"_doc"}}'."\n";
+            $body .= $document['formatted'];
+        }
+
+        $response = $this->client->request('POST', $this->endpoint.'/_bulk', [
+            'body' => $body,
+            'headers' => [
+                'Content-Type' => 'application/json',
+            ],
+        ]);
+
+        $this->responses->attach($response);
+
+        foreach ($this->client->stream($this->responses, 0) as $response => $chunk) {
```

Comment:
this will block until we get some data from the response, while `sendToElasticsearch` is called. So this defeats the non-blocking nature of HttpClient if you expect that it will lead to a non-blocking `log` call.

Reply:
Ah! @nicolas-grekas told me to do that :( Maybe he does not know this component! :joy: 

What I want to achieve:
* be non-blocking
* but Response::desctruct() will block, so I need to keep all response "somewhere"
* from time to time, clean "somewhere"

In my first implementation, I used a fixed array. But this is a bit stupid.
What do you suggest to fix this code ?

(Note: according to  [blackfire](https://blackfire.io/profiles/96195738-7c64-4d90-a2e7-6ac3521f9067/graph), it seems to work...)

Reply:
you spend 23% of the time in this handler, most of it being actually due to calling the HttpClient component. That's not non-blocking.

Should we instead perform this cleaning in `flush` and `close` ?

Reply:
> Should we instead perform this cleaning in `flush` and `close` ?

No because In a CLI context, this may happen not so much.

Reply:
hmm, actually, the fact of passing `0` as timeout in the second argument indeed means that we will process only the chunks which have been received. So this is fine. I missed it initially (I thought about it while reviewing https://github.com/symfony/symfony/pull/32174).